frappe/public/js/frappe/model/create_new.js

```diff
@@ -65,7 +65,7 @@ $.extend(frappe.model, {
 
 					doc[f.fieldname] = v;
 					updated.push(f.fieldname);
-				} else if(f.fieldtype == "Select" && f.options
+				} else if(f.fieldtype == "Select" && f.options && typeof f.options === 'string'
```

Comment:
When we are using `{'value': 'hello', 'label': 'World'}` pattern to define the options for a `Select` field, the above fix, ensure that the next rule `in_list` only will be checked if the `options` are `String`